custom_components/econnect_metronet/__init__.py

```diff
@@ -37,6 +37,20 @@ async def async_migrate_entry(hass, config: ConfigEntry):
         config.version = 2
         hass.config_entries.async_update_entry(config, data=migrated_config)
 
+    if config.version == 2:
```

Comment:
This handles the migration so that we can use the new format without asking users to do any change.